src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -295,4 +297,23 @@ public Action updateAction(final Action action) {
     }
     return existingAction;
   }
+
+  @Transactional
+  public void rerunAction(final List<UUID> actionIds) throws CTPException {
+    for (UUID actionId : actionIds) {
+      Action actionToReRun = actionRepo.findById(actionId);
+      log.with("action_Id", actionId.toString()).debug("Rerunning Aborted Action");
+
+      if (actionToReRun == null) {
+        throw new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND, ACTION_NOT_FOUND, actionIds);
```

Comment:
I would specify the specific action ID that wasn't found here, not the list of actionIDs